### PR TITLE
[batch] improve logging, remove dead code, unify code

### PR DESCRIPTION
**Pull request by danking**
- introduce job logging methods that automatically include id, state, and pod name
- add `reap_job` state update logic into `new_state`, use `new_state` in `reap_job`
- remove unused function `refresh_parents_and_maybe_create`, which lead to more dead code which was also removed

### CODE REVIEW

File: batch/batch/batch.py [hunk 1]
```diff
@@ -129,6 +129,15 @@ class JobStateWriteFailure(Exception):
 
 
 class Job:
+    def _log(self, fun, message, *args, **kwargs):
+        fun(f'{self.id} {self._state} {self._pod_name}: ' + message, *args, **kwargs)
+
+    def log_info(self, message, *args, **kwargs):
+        self._log(log.info, message, *args, **kwargs)
+
+    def log_warning(self, message, *args, **kwargs):
```
Comment (jigold) [inline batch/batch/batch.py]: Add error as well just so it's there and complete.

File: batch/batch/batch.py [hunk 2]
```diff
@@ -279,28 +288,28 @@ def _keep_alive_container(self):  # pylint: disable=R0201
...
         if err is not None:
             if err.status == 409:
-                log.info(f'pod already exists for job {self.id}')
+                self.log_info(f'pod already exists')
                 return
             traceback.print_tb(err.__traceback__)
-            log.info(f'pod creation failed for job {self.id} '
-                     f'with the following error: {err}')
+            self.log_info(
+                f'pod creation failed with the following error: {err}')
```
Comment (jigold) [inline batch/batch/batch.py]: This can be one line now

File: batch/batch/batch.py [hunk 3]
```diff
@@ -341,17 +350,16 @@ def _keep_alive_container(self):  # pylint: disable=R0201
             pod_status, err = await app['k8s'].read_pod_status(self._pod_name, pretty=True)
             if err is not None:
                 traceback.print_tb(err.__traceback__)
-                log.info(f'ignoring: could not get pod status for {self.id} '
-                         f'due to {err}')
+                self.log_info(
+                    f'ignoring: could not get pod status due to {err}')
```
Comment (jigold) [inline batch/batch/batch.py]: one line

File: batch/batch/batch.py [hunk 4]
```diff
@@ -476,41 +484,44 @@ def __init__(self, batch_id, job_id, attributes, callback, userdata, user, alway
...
+                self.exit_codes = exit_codes
             await self.notify_children(new_state)
 
     async def notify_children(self, new_state):
         if new_state not in complete_states:
+            self.log_info(f'{new_state} not complete, will not notify children')
             return
 
         children = [Job.from_record(record) for record in await db.jobs.get_children(*self.id)]
+        self.log_info(f'children: {j.id for j in children}')
```
Comment (jigold) [inline batch/batch/batch.py]: Do we want this?

File: batch/batch/batch.py [hunk 5]
```diff
@@ -655,40 +665,16 @@ def is_successful(self):
...
-
-        self._state = new_state
-
-        await app['pod_throttler'].delete_pod(self)
-        await self._delete_pvc()
-        await self.notify_children(new_state)
-
-        log.info('job {} complete with state {}, exit_codes {}'.format(self.id, self._state, self.exit_codes))
-
+        await self._delete_k8s_resources()
```
Comment (jigold) [inline batch/batch/batch.py]: Shouldn't we set the state before deleting resources?

File: batch/batch/batch.py [hunk 6]
```diff
@@ -617,8 +627,8 @@ def is_successful(self):
                 app['k8s'].read_pod_log(pod.metadata.name, container=container)
                 for container in tasks)))
         if any(err is not None for err in errs):
-            log.info(f'failed to read log for {self.id}, pod '
-                     f'{pod.metadata.name}, {errs} rescheduling ')
+            self.log_info(
```
Comment (jigold) [inline batch/batch/batch.py]: one line

File: batch/batch/batch.py (hunk 2 above)
Comment (danking) [reply batch/batch/batch.py]: it's 81 characters, but I made it one line ;)

File: batch/batch/batch.py (hunk 3 above)
Comment (danking) [reply batch/batch/batch.py]: also 81 characters, but I made it one line

File: batch/batch/batch.py (hunk 4 above)
Comment (danking) [reply batch/batch/batch.py]: yeah, a deploy PR didn't notify its children and I don't know why, this gives us more fine grained information.

Comment (danking) [reply batch/batch/batch.py]: Like, maybe the children list mysteriously is empty?

Comment (jigold) [reply batch/batch/batch.py]: There's a race condition that could have happened. Marked the job complete in the db but then batch gets redeployed before notify children happens. No way to solve this without some sort of trigger cascade in the db.

File: batch/batch/batch.py (hunk 5 above)
Comment (danking) [reply batch/batch/batch.py]: done.

File: batch/batch/batch.py (hunk 4 above)
Comment (danking) [reply batch/batch/batch.py]: Mmm true. We should probably check for children ready to run in the refresh loop. Anyway, in this case, I saw two log statements around notify_children, so it must've run.

**Comment by johnc1231** [comment]
Timing out, might need a bump, or might be broken.

**Pull request merged**